AdaptiveRoads/Data/NetworkExtensions/LaneTransition.cs

```diff
@@ -168,11 +171,27 @@ public enum Flags {
                 var laneInfoD = this.LaneInfoD;
                 bool similar = laneInfoA.m_finalDirection == laneInfoD.m_finalDirection &&
                     laneInfoA.m_laneType == laneInfoD.m_laneType &&
-                    LaneInfoA.m_vehicleType == LaneInfoD.m_vehicleType &&
+                    laneInfoA.m_vehicleType == laneInfoD.m_vehicleType &&
                     laneInfoA.m_similarLaneIndex == laneInfoD.m_similarLaneIndex;
                 m_flags = m_flags.SetFlags(Flags.SimilarLaneIndex, on: similar);
             }
 
+            {
+                var laneInfoA = this.LaneInfoA;
+                var laneInfoD = this.LaneInfoD;
+                var similar = SegmentA.GetHeadNode() == SegmentD.GetTailNode() && //sameDirection
```

Comment:
if road is symmetrical then same direction restriction does not make sense.

Reply:
what about ` SegmentD.GetHeadNode() == SegmentA.GetTailNode() ` ?

Reply:
another way to do this is:
sameDir = `(SegmentA.IsInvert() == SegmentD.IsInvert()) ^ (SegmentA.IsStartNode(NodeId) != SegmentD.IsStartNode(NodeId))`